### ISSUE TITLE
probability output for next token in sequence

### ISSUE THREAD

**Issue by hooande** [minimaxir/gpt-2-simple#168]
I am trying to get the probability of the next token given a set of context tokens. My input is all numbers, ie "0 5 16 21...". I would like to get the probability of each potential number being the next in the sequence. Instead of generating a string of sampled text, I want the logit values for the one next token.

This isn't standard sample generation, so I'm trying to call the model directly in the generate() method of gpt_2.py, like so:

`past=None`
`lm_output = model.model(hparams=hparams, X=context, past=past, reuse=tf.compat.v1.AUTO_REUSE)`
`out = sess.run(lm_output, feed_dict={context: [context_tokens]})`

then accessing the probabilities with

`probs = sess.run(tf.nn.softmax(out['logits'][0][0]))`

but it seems to be ignoring the context tokens and generating from the beginning of a sequence. I think I might need to set values in the 'past' tensor? but I'm not sure how to do that.

Any help would be greatly appreciated!

**Comment by inspire22** [comment]
I was thinking of trying to do something like this in order to add a fitness function to the weighting of possible outputs.  Kinda like hacking top_k.  Did you ever get this working, or have any suggestions please?
